### ISSUE TITLE
Obsidian freezes when title or alias have special characters.

### ISSUE THREAD

**Issue by reemagit** [snezhig/obsidian-front-matter-title#154]
Hello, I am using the plugin to visualize the title or the alias instead of the filename everywhere, including the links to the file. However, when the title or alias contain some special characters such as the dash "-" or even colon ":", I have some critical issue. 
However, whenever I have special characters in the title or alias (even just dash "-" or colon ":") and I link the file somewhere else with [[TITLE]] or [[ALIAS]], after a few keystrokes Obsidian freezes completely. 
I tried to check if I have output output even in the debug console, but I found none, the whole program is just frozen. I can only terminate the process and restart Obsidian.

I couldn't find anyone else having this problem, so I am not sure if it depends on something else, yet when I link the file with the filename everything works as normal. 

Am I the only one having this issue? Thanks.

**Comment by snezhig** [comment]
Hi, Can you give me your plugin settings and some examples of documents to reproduce it?
Does it happen if plugin is disabled?

**Comment by reemagit** [comment]
Hello, thanks for answering. 

> Can you give me your plugin settings and some examples of documents to reproduce it?

I am not sure if there's a more comprehensive way to list the settings, but I enabled all the items in the "Features" section. As the common main template I have "title" and for common fallback template I have "fallback_title". Rest should be default.

> Does it happen if plugin is disabled?

If I disable it it stops happening, even if I leave the same link to to the page with the dash. 
However, I noticed additional patterns: for some other notes with a dash ("-") in the title, if I link to them in a note I don't get a complete freeze, but Obsidian hangs for some seconds every couple of seconds. For some other notes everything works, regardless of whether they have a dash or not. 

To summarize:
- If the plugin is enabled, and I link a note with a dash ("-") in the title, it either (1) freezes, (2) hangs periodically, or (3) nothing happens, depending on which is the note that I am linking
- If the plugin is disabled this doesn't seem to happen
- If I remove the dash from the linked note title this doesn't seem to happen either.

Sorry if this may not be super clear, but it's the best description I can provide of the situation.

Please let me know if I can provide other info.

Thanks.

**Comment by snezhig** [comment]
@reemagit  I suppose that the cause is Note Link feature. You can disable it until I fix the problem.

**Comment by reemagit** [comment]
Sounds good, thanks.

**Comment by snezhig** [comment]
Should be fixed in 3.5.3

**Comment by reemagit** [comment]
Seems to be fixed, thank you very much.
